pkg/apiserver/event/event.go

```diff
@@ -56,30 +56,74 @@ func Register(r *gin.RouterGroup, s *Service) {
-// @Description Get all chaos events from db.
+// @Summary Get the list of events from db.
+// @Description Get the list of events from db.
 // @Tags events
 // @Produce json
+// @Param podName query string false "The pod's name"
+// @Param podNamespace query string false "The pod's namespace"
+// @Param startTime query string false "The start time of events"
+// @Param endTime query string false "The end time of events"
+// @Param experimentName query string false "The name of the experiment"
+// @Param experimentNamespace query string false "The namespace of the experiment"
+// @Param uid query string false "The UID of the experiment"
+// @Param kind query string false "kind" Enums(PodChaos, IoChaos, NetworkChaos, TimeChaos, KernelChaos, StressChaos)
 // @Success 200 {array} core.Event
 // @Router /api/events [get]
 // @Failure 500 {object} utils.APIError
 func (s *Service) listEvents(c *gin.Context) {
-	podName := c.Query("podName")
-	podNamespace := c.Query("podNamespace")
-	startTimeStr := c.Query("startTime")
-	experimentName := c.Query("experimentName")
-	experimentNamespace := c.Query("experimentNamespace")
-	uid := c.Query("uid")
-
-	if podName != "" && podNamespace == "" {
+	filter := core.Filter{
+		PodName:             c.Query("podName"),
+		PodNamespace:        c.Query("podNamespace"),
+		StartTimeStr:        c.Query("startTime"),
+		FinishTimeStr:       c.Query("finishTime"),
+		ExperimentName:      c.Query("experimentName"),
+		ExperimentNamespace: c.Query("experimentNamespace"),
+		UID:                 c.Query("uid"),
+		Kind:                c.Query("kind"),
+	}
+
+	if filter.PodName != "" && filter.PodNamespace == "" {
 		c.Status(http.StatusInternalServerError)
 		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(fmt.Errorf("when podName is not empty, podNamespace cannot be empty")))
 		return
 	}
 
-	eventList, err := s.event.ListByFilter(context.Background(), podName, podNamespace, experimentName, experimentNamespace, uid, startTimeStr)
+	eventList, err := s.event.ListByFilter(context.Background(), filter)
+	if err != nil {
+		c.Status(http.StatusInternalServerError)
+		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
+		return
+	}
+
+	c.JSON(http.StatusOK, eventList)
+}
+
+// @Summary Get the list of events without pod records from db.
+// @Description Get the list of events without pod records from db.
+// @Tags events
+// @Produce json
+// @Param startTime query string false "The start time of events"
+// @Param endTime query string false "The end time of events"
+// @Param experimentName query string false "The name of the experiment"
+// @Param experimentNamespace query string false "The namespace of the experiment"
+// @Param kind query string false "kind" Enums(PodChaos, IoChaos, NetworkChaos, TimeChaos, KernelChaos, StressChaos)
+// @Success 200 {array} core.Event
+// @Router /api/events/dry [get]
+// @Failure 500 {object} utils.APIError
+func (s *Service) listDryEvents(c *gin.Context) {
```

Comment:
It is not necessary to have a new API for this. One additional parameter is enough.